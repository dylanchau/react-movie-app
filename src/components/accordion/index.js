import { createContext, useContext, useState } from 'react'

import {
  Body,
  Container,
  Frame,
  Header,
  Inner,
  Item,
  Title,
} from './styles/accordion'

const ToggleContext = createContext()

export default function Accordion({ children, ...restProps }) {
  return (
    <Container {...restProps}>
      <Inner>{children}</Inner>
    </Container>
  )
}

Accordion.Title = function AccordionTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>
}

Accordion.Frame = function AccordionFrame({ children, ...restProps }) {
  return <Frame {...restProps}>{children}</Frame>
}

Accordion.Item = function AccordionItem({ children, ...restProps }) {
  const [toggle, setToggle] = useState(false)

  return (
    <ToggleContext.Provider value={{ toggle, setToggle }}>
      <Item {...restProps}>{children}</Item>
    </ToggleContext.Provider>
  )
}

Accordion.Header = function AccordionHeader({ children, ...restProps }) {
  const { toggle, setToggle } = useContext(ToggleContext)

  return (
    <Header onClick={() => setToggle((preState) => !preState)} {...restProps}>
      {children}
      {toggle ? (
        <img src="/images/icons/close-slim.png" alt="Close" />
      ) : (
        <img src="/images/icons/add.png" alt="Open" />
      )}
    </Header>
  )
}

Accordion.Body = function AccordionBody({ children, ...restProps }) {
  const { toggle } = useContext(ToggleContext)
  return toggle ? <Body {...restProps}>{children}</Body> : null
}
