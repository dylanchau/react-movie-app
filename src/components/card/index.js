import { createContext, useContext, useState } from 'react'

import {
  Container,
  Content,
  Entities,
  Feature,
  FeatureClose,
  FeatureText,
  FeatureTitle,
  Group,
  Image,
  Item,
  Maturity,
  Meta,
  SubTitle,
  Text,
  Title,
} from './styles/card'

export const FeatureContext = createContext()

export default function Card({ children, ...restProps }) {
  const [showFeature, setShowFeature] = useState(false)
  const [itemFeature, setItemFeature] = useState({})

  return (
    <FeatureContext.Provider
      value={{ showFeature, setShowFeature, itemFeature, setItemFeature }}
    >
      <Container {...restProps}>{children}</Container>
    </FeatureContext.Provider>
  )
}

Card.Group = function CardGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>
}

Card.Tittle = function CardTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>
}

Card.SubTittle = function CardSubTittle({ children, ...restProps }) {
  return <SubTitle {...restProps}>{children}</SubTitle>
}

Card.Text = function CardText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>
}

Card.Meta = function CardMeta({ children, ...restProps }) {
  return <Meta {...restProps}>{children}</Meta>
}

Card.Image = function CardImage({ children, ...restProps }) {
  return <Image {...restProps} />
}

Card.Item = function CardItem({ item, children, ...restProps }) {
  const { setItemFeature, setShowFeature } = useContext(FeatureContext)

  return (
    <Item
      onClick={() => {
        setItemFeature(item)
        setShowFeature(true)
      }}
      {...restProps}
    >
      {children}
    </Item>
  )
}

Card.Entities = function CardEntities({ children, ...restProps }) {
  return <Entities {...restProps}>{children}</Entities>
}

Card.Entities = function CardEntities({ children, ...restProps }) {
  return <Entities {...restProps}>{children}</Entities>
}

Card.Feature = function CardFeature({ children, category, ...restProps }) {
  const { showFeature, itemFeature, setShowFeature } = useContext(
    FeatureContext
  )
  return showFeature ? (
    <Feature
      src={`/images/${category}/${itemFeature.genre}/${itemFeature.slug}/large.jpg`}
      {...restProps}
    >
      <Content>
        <FeatureTitle>{itemFeature.title}</FeatureTitle>
        <FeatureText>{itemFeature.description}</FeatureText>
        <FeatureClose onClick={() => setShowFeature((prev) => !prev)}>
          <Image src="/images/icons/close.png" alt="Close" />
        </FeatureClose>

        <Group margin="30px 0" flexDirection="row" alginItem="center">
          <Maturity rating={itemFeature.maturity}>
            {itemFeature.maturity < 12 ? 'PG' : itemFeature.maturity}
          </Maturity>
          <FeatureText fontWeight="bold">
            {itemFeature.genre.charAt(0).toUpperCase() +
              itemFeature.genre.slice(1)}
          </FeatureText>
        </Group>
        {children}
      </Content>
    </Feature>
  ) : null
}
