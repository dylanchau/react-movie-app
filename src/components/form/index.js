import {
  Base,
  Container,
  Error,
  Input,
  Link,
  Submit,
  Text,
  TextSmall,
  Title,
} from './styles/form'

export default function Form({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>
}

Form.Title = ({ children, ...restProps }) => (
  <Title {...restProps}>{children}</Title>
)

Form.Base = ({ children, ...restProps }) => (
  <Base {...restProps}>{children}</Base>
)

Form.Text = ({ children, ...restProps }) => (
  <Text {...restProps}>{children}</Text>
)

Form.TextSmall = ({ children, ...restProps }) => (
  <TextSmall {...restProps}>{children}</TextSmall>
)

Form.Submit = ({ children, ...restProps }) => (
  <Submit {...restProps}>{children}</Submit>
)

Form.Error = ({ children, ...restProps }) => (
  <Error {...restProps}>{children}</Error>
)

Form.Link = ({ children, ...restProps }) => (
  <Link {...restProps}>{children}</Link>
)

Form.Input = ({ children, ...restProps }) => (
  <Input {...restProps}>{children}</Input>
)
