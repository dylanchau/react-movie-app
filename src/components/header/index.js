import { useState } from 'react'
import { Link as ReactRouterLink } from 'react-router-dom'

import {
  Background,
  ButtonLink,
  Container,
  Dropdown,
  Feature,
  FeatureCallOut,
  Group,
  Link,
  Logo,
  Picture,
  PlayButton,
  Profile,
  Search,
  SearchIcon,
  SearchInput,
  Text,
} from './styles/header'

export default function Header({ bg = true, children, ...restProps }) {
  return bg ? <Background {...restProps}>{children}</Background> : children
}

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>
}

Header.Group = function HeaderGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>
}

Header.Search = function HeaderSearch({
  searchTerm,
  setSearchTerm,
  ...restProps
}) {
  const [searchActive, setSearchActive] = useState(false)

  return (
    <Search {...restProps}>
      <SearchIcon
        onClick={() => setSearchActive((preProp) => !preProp)}
        data-testid="search-click"
      >
        <img alt="search" src="/images/icons/search.png" />
      </SearchIcon>
      <SearchInput
        value={searchTerm}
        onChange={({ target }) => setSearchTerm(target.value)}
        active={searchActive}
        placeholder="Search..."
        data-testid="search-input"
      />
    </Search>
  )
}

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>
}

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  )
}

Header.Feature = function HeaderFeature({ children, ...restProps }) {
  return <Feature {...restProps}>{children}</Feature>
}

Header.FeatureCallOut = function HeaderFeatureCallOut({
  children,
  ...restProps
}) {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>
}

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>
}

Header.TextLink = function LinkText({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>
}

Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>
}

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>
}

Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return <Picture {...restProps} src={`/images/users/${src}.png`} />
}

Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
  return <PlayButton {...restProps}>{children}</PlayButton>
}
