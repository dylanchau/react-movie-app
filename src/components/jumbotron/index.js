import { Inner } from './styles/jumbotron';

export default function Jumbotron() {
  return (
    <Inner>
      <p>Hello</p>
    </Inner>
  );
}
