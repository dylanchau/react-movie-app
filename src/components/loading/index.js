import { LockBody, Picture, ReleaseBody, Spinner } from './styles/loading'

function Loading({ src, ...restProps }) {
  return (
    <Spinner {...restProps}>
      <LockBody />
      <Picture src={`/images/users/${src}.png`} alt={src} />
    </Spinner>
  )
}

Loading.ReleaseBody = function LoadingReleaseBody({ ...restProps }) {
  return <ReleaseBody {...restProps} />
}

export default Loading
