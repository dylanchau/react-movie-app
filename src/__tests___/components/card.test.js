import { fireEvent, render } from '@testing-library/react'
import { Card, Player } from 'components'
import React from 'react'

const category = 'series'

const slideRows = [
  {
    title: 'Documentaries',
    data: [
      {
        genre: 'documentaries',
        maturity: '18',
        slug: 'tiger-king',
        description: 'Tiger King description',
        id: '07adcf19-517f-41b4-8f8b-04fee694bce1',
        title: 'Tiger King',
        docId: 'IcRxJBanzE7nmIp2GMGb',
      },
    ],
  },
  {
    title: 'Feel Good',
    data: [
      {
        title: 'Juno',
        genre: 'feel-good',
        description: 'Juno description',
        maturity: '0',
        id: 'a938b5a6-fe25-4a06-8050-353bc7146ccd',
        slug: 'juno',
        docId: '4JDgdf5vE0K3YrW9ZnbP',
      },
    ],
  },
]

const Default = () => (
  <Card.Group>
    {slideRows.map((slide) => (
      <Card key={`${category}-${slide.title.toLowerCase()}`}>
        <Card.Tittle>{slide.title}</Card.Tittle>
        <Card.Entities>
          {slide.data.map((item) => (
            <Card.Item key={item.docId} item={item} data-testid={item.docId}>
              <Card.Image
                src={`/images/${category}/${item.genre}/${item.slug}/small.jpg`}
              />
              <Card.Meta>
                <Card.SubTittle>{item.title}</Card.SubTittle>
                <Card.Text>{item.description}</Card.Text>
              </Card.Meta>
            </Card.Item>
          ))}
        </Card.Entities>

        <Card.Feature category={category}>
          <Player>
            <Player.Button />
            <Player.Video src="/videos/bunny.mp4" />
          </Player>
        </Card.Feature>
      </Card>
    ))}
  </Card.Group>
)

describe('<Card /> component', () => {
  it('Render card component and toggle card item', () => {
    const {
      container,
      getByText,
      queryByTestId,
      getByAltText,
      queryByText,
    } = render(<Default />)

    expect(getByText('Documentaries')).toBeTruthy()
    expect(getByText('Feel Good')).toBeTruthy()

    fireEvent.mouseOver(queryByTestId('IcRxJBanzE7nmIp2GMGb'))
    expect(getByText('Tiger King')).toBeTruthy()
    expect(getByText('Tiger King description')).toBeTruthy()
    fireEvent.mouseOver(queryByTestId('4JDgdf5vE0K3YrW9ZnbP'))
    expect(getByText('Juno')).toBeTruthy()
    expect(getByText('Juno description')).toBeTruthy()

    fireEvent.click(queryByTestId('IcRxJBanzE7nmIp2GMGb'))
    expect(getByText('18')).toBeTruthy()
    fireEvent.click(getByAltText('Close'))
    expect(queryByText('18')).toBeFalsy()

    fireEvent.click(queryByTestId('4JDgdf5vE0K3YrW9ZnbP'))
    expect(getByText('PG')).toBeTruthy()
    fireEvent.click(getByAltText('Close'))
    expect(queryByText('PG')).toBeFalsy()

    expect(container.firstChild).toMatchSnapshot()
  })
})
