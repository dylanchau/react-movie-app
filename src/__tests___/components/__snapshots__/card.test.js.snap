// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> component Render card component and toggle card item 1`] = `
<div
  class="card__Group-sc-9frf9s-2 cKFQUF"
>
  <div
    class="card__Container-sc-9frf9s-1 lijfTr"
  >
    <p
      class="card__Title-sc-9frf9s-0 eovtZC"
    >
      Documentaries
    </p>
    <div
      class="card__Entities-sc-9frf9s-5 dmgLgZ"
    >
      <div
        class="card__Item-sc-9frf9s-8 RcSzS"
        data-testid="IcRxJBanzE7nmIp2GMGb"
      >
        <img
          class="card__Image-sc-9frf9s-7 gjkdWl"
          src="/images/series/documentaries/tiger-king/small.jpg"
        />
        <div
          class="card__Meta-sc-9frf9s-6 fijcrh"
        >
          <p
            class="card__SubTitle-sc-9frf9s-3 iEsgdO"
          >
            Tiger King
          </p>
          <p
            class="card__Text-sc-9frf9s-4 csZgrT"
          >
            Tiger King description
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="card__Container-sc-9frf9s-1 lijfTr"
  >
    <p
      class="card__Title-sc-9frf9s-0 eovtZC"
    >
      Feel Good
    </p>
    <div
      class="card__Entities-sc-9frf9s-5 dmgLgZ"
    >
      <div
        class="card__Item-sc-9frf9s-8 RcSzS"
        data-testid="4JDgdf5vE0K3YrW9ZnbP"
      >
        <img
          class="card__Image-sc-9frf9s-7 gjkdWl"
          src="/images/series/feel-good/juno/small.jpg"
        />
        <div
          class="card__Meta-sc-9frf9s-6 fijcrh"
        >
          <p
            class="card__SubTitle-sc-9frf9s-3 iEsgdO"
          >
            Juno
          </p>
          <p
            class="card__Text-sc-9frf9s-4 csZgrT"
          >
            Juno description
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
